Feature: OAuth 가입 & 보호소 등록 기능 #14

HuttTheJAVA opened the pull request:
# Feature: OAuth 가입 & 보호소 등록 기능 #14

### ✅ PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [x] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### ✅ 개요
>  일반 회원 OAuth 가입 구현 (KAKAO,NAVER)
 보호소의 등록 기능 구현
 OAuth 로그인 성공 시 JWT 발급 기능

### ✅ 리뷰어가 꼭 봐줬으면 하는 부분
> 양이 좀 많습니다... 리뷰하는데 이해 안되는 점 있으면 바로 말씀해주세요😅

### ✅ ISSUE 번호
- #14 

### ✅ 참고 사항
> 총 7가지의 설정 값 추가가 필요합니다. (코드 상 "${~~~}" 로 돼 있는 애들)
1. GCSconfig 클래스의 keyFileLocation 값
2. GoogleCloudStorageUploader의 bucketName
3. KakaoHttpClient 클래스의 clientId & redirectUrl
4. NaverHttpClient 클래스의 clientId & clientSecret & redirectUri

3번,4번에 해당하는 값은 따로 공유하겠습니다.
1번,2번은 GCS의 설정 값이 어떻게 구성됐는지 확인해 봐야 할 것 같습니다.

HuttTheJAVA committed:
chore: build.gradle 의존성 추가 #14
- FeignClient를 위한 OpenFeign 의존성 추가
- Google Cloud Storage 의존성 추가

HuttTheJAVA committed:
feat: 커스텀 예외 구현 및 GlobalExceptionHandler에 예외처리 추가 #14
- OAuthProcessingException 커스텀 예외 구현
- email 중복, nickname 중복에 던져지는 DuplicateEntityException 구현
- GlobalExceptionHandler에서 커스텀 예외 처리 구현

HuttTheJAVA committed:
feat: 보호소 및 유저 등록 기능 구현 #14
- 보호소 등록에 필요한 Controller, Service, Repository 구현
- 유저 등록에 필요한 Controller, Service, Repository 구현
- 보호소, 유저 등록에 필요한 부가적인 기능 구현

HuttTheJAVA committed:
feat: OAuth 회원 가입 기능 구현 #14
- Naver OAuth 가입 기능 구현
- Kakao OAuth 가입 기능 구현

sngbmYu commented on src/main/java/com/jaefan/munpyspring/user/presentation/UserController.java:
RequestMapping을 /api/users로 바꾸는 게 좋을 것 같습니다.
UserController의 매핑이 명확해지고, refresh나 logout 메서드가 User 도메인에 속했음을 URL을 통해 쉽게 확인할 수 있습니다.

HuttTheJAVA replied:
수정 완료!👍

sngbmYu commented on src/main/java/com/jaefan/munpyspring/shelter/presentation/ShelterController.java:
RequestMapping을 /api/shelters로 바꾸는 게 좋을 것 같습니다.
ShelterController의 매핑이 명확해지고, 이후 메서드가 추가되더라도 명확하게 shelter 도메인임을 나타낼 수 있습니다.

HuttTheJAVA replied:
수정 완료!👍

sngbmYu commented on src/main/java/com/jaefan/munpyspring/user/presentation/dto/RefreshTokenRequest.java:
이것도 클래스명에 Dto 붙여주실 수 있나요? 통일시키는 게 좋을 것 같습니다!

HuttTheJAVA replied:
수정 완료!👍

sngbmYu commented on src/main/java/com/jaefan/munpyspring/user/presentation/dto/UserSignUpRequestDto.java:
toUser 메서드를 개선할 수 있을 것 같습니다.

해당 메서드를 사용하는 코드를 확인해보았는데, 파라미터를 통해 생성된 UserSignUpRequestDto를 그대로 인자로 받고 있네요.
그렇다면 static으로 만들어서 인스턴스 자체를 인자로 받을 필요 없이, 일반 메서드로 만들어서 인스턴스 자신이 가지고 있는 데이터를 사용하는 것이 더 좋을 것 같습니다.

그러면 UserService에서도 UserSignUpRequestDto.toUser(userSignUpRequestDto, ...)이 아니라, userSignUpRequestDto.toUser(...)처럼 사용할 수 있습니다. 이쪽이 더 간결한 것 같네요.

혹시 Service 외에도 사용처가 있어서, static이 필요한 경우라면 말씀해주세요.

HuttTheJAVA replied:
말씀하신대로 굳이 static으로 할 필요는 없을 것 같습니다. 말씀하신 사항 반영하겠습니다!

HuttTheJAVA replied:
수정 완료!👍

HuttTheJAVA committed:
refactor: Controller RequestMapping 수정 및 dto 클래스명 수정 #14
- ShelterController와 UserController의 RequestMapping 수정
- RefreshTokenRequest -> RefreshTokenRequestDto로 클래스명 수정
- OAuthToken -> OAuthTokenDto
- UserSignUpRequestDto의 toUser 메소드의 static 제거

HuttTheJAVA committed:
refactor: Shelter 및 User Dto 클래스의 toEntity 메소드 수정 #14
- shelterSignUpRequestDto의 toShelter 수정
- UserSignUpRequestDto의 toUser 내부 빌더 메소드에 필드 추가

sngbmYu reviewed (approved):
LGTM 👍
고생하셨습니다!

GoByeonghu reviewed (approved):
LGTM입니다:)

Namgyu11 reviewed (approved):
고생 많으셨습니다!!
LGTM😎😎👍
